### PR TITLE
Possibility to skip commit creation (but keep a tag)

### PR DESCRIPTION
**Pull request by cakeinpanic**
Hi!
In our company we have a huge monorepos with multiple teams working on them.
I want to use semver to provide autorelease for them, but when it comes to _pushing created commits back to master_ from the CI job, we face a huge **concurrency issue.**

Like if two people merge their PRs at the same time and each job needs to push back to master `chore` commits, one of them would definitely fail

To avoid this situation we decided **not to update package version** in the code, but only put git tags + **store changelog in the github releases**. This does not sound like a best practice but we got what we got

To achieve that I use custom script which moves latest tag on a previous commit after `semver` finishes.

This PR provides an additional `--skipCommit` flag so commit can be skipped initially without custom scripts

### CODE REVIEW

**Comment by cakeinpanic** [comment]
Apparently some changes needs to be added: tag should be put not on the last commit, but on last commit in exact _library_, otherwise next time process can be broken. I'd add this functionality

File: packages/semver/src/executors/version/index.ts [hunk 1]
```diff
@@ -227,6 +229,7 @@ function _normalizeOptions(options: VersionBuilderSchema) {
     changelogHeader: options.changelogHeader ?? defaultHeader,
     versionTagPrefix: options.tagPrefix ?? options.versionTagPrefix,
     commitMessageFormat: options.commitMessageFormat as string,
+    skipCommit: options.skipCommit || false,
```
Comment (edbzn) [inline packages/semver/src/executors/version/index.ts:232]: The default value should be placed in the `schema.json`, here you can type assert `options.skipCommit as boolean`.

**Review by edbzn — COMMENTED** [review]
Looks good overall.

> Apparently some changes needs to be added: tag should be put not on the last commit, but on last commit in exact library, otherwise next time process can be broken. I'd add this functionality

Yes, good catch!

File: packages/semver/src/executors/version/utils/commit.ts [hunk 2]
```diff
@@ -6,18 +7,24 @@ import { createTemplateString } from './template-string';
...
   projectName: string;
-}): Observable<void> {
+}): Observable<void | null> {
   if (dryRun) {
     return EMPTY;
   }
 
+  if (skipCommit) {
+    return of(null);
+  }
```
[the remark this answers is not in the document]
Comment (cakeinpanic) [reply packages/semver/src/executors/version/utils/commit.ts]: fixed!

File: packages/semver/src/executors/version/index.ts (hunk 1 above)
Comment (cakeinpanic) [reply packages/semver/src/executors/version/index.ts:235]: fixed

**Comment by cakeinpanic** [comment]
@edbzn hi, added logic changes mentioned above, please take a look

File: packages/semver/src/executors/version/schema.json [hunk 3]
```diff
@@ -89,6 +89,11 @@
       "type": "boolean",
       "default": false
     },
+    "skipCommit": {
+      "description": "Allows to skip making a commit when bumping a version(but put the tag on last existent commit)",
```
Comment (edbzn) [inline packages/semver/src/executors/version/schema.json:93]:
```suggestion
      "description": "Allows to skip making a commit when bumping a version and put the tag on last existent commit",
```

File: packages/semver/src/executors/version/utils/git.ts [hunk 4]
```diff
@@ -5,26 +5,52 @@ import { exec } from '../../common/exec';
...
 }: {
   projectRoot: string;
   since: string;
+}): Observable<string[]> {
+  return getFormattedCommits({since, projectRoot, format:'%B'})
+}
+/**
+ * Return hash of last commit of a project
+ */
+export function getLastProjectCommitHash({
```
Comment (edbzn) [inline packages/semver/src/executors/version/utils/git.ts:22]: I would rename it to `getCommitHash`, the param is explicit so we don't have to specify it in the function name IMO.

File: packages/semver/src/executors/version/utils/git.ts [hunk 5]
```diff
@@ -5,26 +5,52 @@ import { exec } from '../../common/exec';
...
+}
+
+function getFormattedCommits({
+  projectRoot,
+  format,
+  since,
+}: {
+  projectRoot: string;
+  format: string,
+  since: string;
```
Comment (edbzn) [inline packages/semver/src/executors/version/utils/git.ts:38]: What about making the since `param` optional?

**Review by edbzn — COMMENTED** [review]
Looks good overall, I left a few comments on minor things. One last thing, could you run prettier over the project? Thanks for the work!

**Comment by edbzn** [comment]
Merged into main, thanks @cakeinpanic for the work!

**Comment by cakeinpanic** [comment]
Wow, I was not notified about this PR, glad you merged it!

**Pull request closed without merging**